tests/test_yroom.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from anyio import TASK_STATUS_IGNORED, sleep
+from anyio.abc import TaskStatus
+from pycrdt import Map
+
+from pycrdt_websocket import exception_logger
+from pycrdt_websocket.yroom import YRoom
+
+pytestmark = pytest.mark.anyio
+
+
+@pytest.mark.parametrize("websocket_server_api", ["websocket_server_start_stop"], indirect=True)
+@pytest.mark.parametrize("yws_server", [{"exception_handler": exception_logger}], indirect=True)
+async def test_yroom_restart(yws_server, yws_provider):
+    port, server = yws_server
+    yroom = YRoom(exception_handler=exception_logger)
+
+    async def raise_error(task_status: TaskStatus[None] = TASK_STATUS_IGNORED):
+        task_status.started()
+        raise RuntimeError("foo")
+
+    yroom.ydoc = yws_provider
+    await server.start_room(yroom)
+    yroom.ydoc["map"] = ymap1 = Map()
+    ymap1["key"] = "value"
+    await yroom._task_group.start(raise_error)
+    ymap1["key2"] = "value2"
```

Comment:
1. Can we add something here to assert that het `raise_error` callback was called? 
2. Can we capture the logs and assert that the YRoom was restarted from those logs?

Reply:
Addressed this way. 
```
task_group_1 = yroom._task_group
await yroom._task_group.start(raise_error)
...
assert yroom._task_group is not task_group_1
```

Reply:
:+1: